core/registry.js

```diff
@@ -126,8 +126,9 @@ Blockly.registry.register = function(type, name, registryItem) {
   // Validate that the given class has all the required properties.
   Blockly.registry.validate_(type, registryItem);
 
-  // If the name already exists throw an error.
-  if (typeRegistry[name]) {
+  // Don't throw an error if it is a toolbox item. We expect developers to override
+  // categories and separators.
+  if (type != Blockly.registry.Type.TOOLBOX_ITEM && typeRegistry[name]) {
```

Comment:
This seems reasonable, but if it ends up being a pattern we should find a good way to distinguish between things we expect to be overridden vs things that should be unique. (That is, if this turns into a list of types that can be overridden, let's find a better structure for the check).